### ISSUE TITLE
GMusicBackend does not implement playlists.as_list(). Please upgrade it

### ISSUE THREAD

**Issue by jonsxaero** [hechtus/mopidy-gmusic#84]
I have a Warning after gmusic authenticated

```

INFO     2015-05-07 12:37:39,680 [4942:GMusicBackend-5] requests.packages.urllib3.connectionpool
  Starting new HTTPS connection (1): play.google.com
INFO     2015-05-07 12:37:40,382 [4942:GMusicBackend-5] gmusicapi.Mobileclient1
  authenticated
INFO     2015-05-07 12:37:40,434 [4942:GMusicBackend-5] requests.packages.urllib3.connectionpool
  Starting new HTTPS connection (1): www.googleapis.com
WARNING  2015-05-07 12:37:43,032 [4942:MainThread] mopidy.core.playlists
  GMusicBackend does not implement playlists.as_list(). Please upgrade it.
INFO     2015-05-07 12:37:43,049 [4942:MainThread] mopidy.commands
  Starting Mopidy frontends: MpdFrontend, HttpFrontend
INFO     2015-05-07 12:37:43,069 [4942:MainThread] mopidy.mpd.actor
  MPD server running at [::ffff:0.0.0.0]:6600
DEBUG    2015-05-07 12:37:43,084 [4942:MainThread] mopidy.http.actor
  Starting HTTP server
DEBUG    2015-05-07 12:37:43,110 [4942:MpdFrontend-11] mopidy.zeroconf
  Zeroconf service _mpd._tcp at [raspberrypi.local]:6600: Avahi service not running; publish failed.
INFO     2015-05-07 12:37:43,114 [4942:HttpFrontend-13] mopidy.http.actor
  HTTP server running at [::ffff:0.0.0.0]:6680
```

This does not allow to make streaming from gmusic. I upgraded mopidy and gmusic to last version, but I still get this warning. I run in raspberry pi in raspbian flavor.

```
pi@raspberrypi ~ $ sudo mopidyctl --version
Running "/usr/bin/mopidy --config /usr/share/mopidy/conf.d:/etc/mopidy/mopidy.conf --version" as user mopidy
Mopidy 1.0.4
pi@raspberrypi ~ $ pip show mopidy-gmusic
---
Metadata-Version: 1.1
Name: Mopidy-GMusic
Version: 0.3.0
Summary: Google Play Music extension for Mopidy
Home-page: http://github.com/hechtus/mopidy-gmusic/
Author: Ronald Hecht
Author-email: ronald.hecht@gmx.de
License: Apache License, Version 2.0
Location: /usr/local/lib/python2.7/dist-packages
Requires: setuptools, Mopidy, Pykka, gmusicapi
Entry-points:
  [mopidy.ext]
  gmusic = mopidy_gmusic:GMusicExtension
```

**Comment by JoeSzymkowicz** [comment]
I also got this message after I used "easy_install mopidy-gmusic" and while the console says that the says the extension is enabled, I do not see any of my playlists or Google Music functionality enabled in the front-end. For all intents and purposes, the extension seems to be completely broken using this install command. I uninstalled.

I then saw a suggestion to try "pip install https://github.com/hechtus/mopidy-gmusic/archive/develop.zip" and the install went fine. I checked deps and everything was satisfied and the both gmusic and it's backend were enabled; Mopidy opened fine. I notice that in "Browse" Google Music Artists and Albums never seem to load. I can open Playlists from Google Music, but not all tracks within the playlist are included. When I play a track that is displayed, the track seems to play fine, and the following message is displayed: "WARNING  Element doesn't implement handling of this stream. Please file a bug." When I try to open an Album from my Google Music Collection, it never seems to load.

Using browse from the web front-end, I am not able to open any Albums, including an Album that contains only a track that I am able to play via Playlist. Browsing Tracks, I am able to only see certain tracks from Albums. The Tracks that I am able to see, I can play. In Browse, when I click on an Artist, the Loading message never goes away unless I back or start another operation.

**Comment by jonsxaero** [comment]
I do the same and got similar results. I think that gmusicapi is outdated and probably many things changed or will change. Last commit was on 22 Mar, and author noted that the project is in low priority since Nov 2014.

**Comment by jonsxaero** [comment]
I try with gmusicapi and when used search_all_access function thown an error:

```
2015-05-12 11:46:25,413 - gmusicapi.Mobileclient1 (shared:281) [ERROR]: the response format for Search was not recognized.

Value {u'rating': u'5', u'albumArtRef': [{u'url': u'http://lh5.ggpht.com/OqTg6K_jAUAs-LYn67b6rJmKRg5ffCf1OTskGinetuwMyDWaOVGVMH0F8jx79_wVdoaU                           Czr-'}, {u'url': u'http://lh3.googleusercontent.com/Cmtq4jndQhoHw-1rxFgqo7UEt_DgGbEfNeehzBq3xra9-H4OpWdHmJnJ_YRJfI_ANEhHOTrA'}], u'lastRating                           ChangeTimestamp': u'1429192790662000', u'artistId': [u'As4u2b6l3sqyfyn2kffo276dfny'], u'composer': u'', u'year': 1992, u'trackAvailableForSub                           scription': True, u'trackType': u'7', u'album': u'Vulgar Display Of Power', u'title': u'Walk', u'albumArtist': u'Pantera', u'trackNumber': 3,                            u'discNumber': 1, u'albumAvailableForPurchase': True, u'contentType': u'2', u'trackAvailableForPurchase': True, u'storeId': u'T4dolo7hxcyzhc                           bo42dhvkp4fyi', u'nid': u'T4dolo7hxcyzhcbo42dhvkp4fyi', u'estimatedSize': u'12607765', u'albumId': u'Boqthnakwoo4scuy7ivt3rp6l54', u'genre':                            u'Metal', u'playCount': 2, u'kind': u'sj#track', u'artist': u'Pantera', u'durationMillis': u'315000'} for field '<obj>.entries[64].track' con                           tains additional property 'lastRatingChangeTimestamp' not defined by 'properties' or 'patternProperties' and additionalProperties  is False
```

The script.py is

```
from gmusicapi import Mobileclient
from gmusicapi._version import __version__

print __version__
api = Mobileclient()
print 'api is',api.login('mail@gmail.com', 'pass')
library = api.search_all_access('Pantera')

```

After uninstall gmusicapi, I install develop version with  pip install https://github.com/simon-weber/Unofficial-Google-Music-API/archive/develop.zip
Then search_all_access goes well. I have not tried mopidy yet.

**Comment by JoeSzymkowicz** [comment]
I have uninstalled both gmusic and gmusicapi packages. I then installed both from the develop branches using the following commands:

pip install https://github.com/simon-weber/Unofficial-Google-Music-API/archive/develop.zip
pip install https://github.com/hechtus/mopidy-gmusic/archive/develop.zip

It resulted in the following error:

Exception in thread Thread-10:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 552, in __bootstrap_inner
    self.run()
  File "/usr/local/lib/python2.7/dist-packages/Mopidy_GMusic-1.0.0-py2.7.egg/mopidy_gmusic/repeating_timer.py", line 12, in run
    self._method()
  File "/usr/local/lib/python2.7/dist-packages/Mopidy_GMusic-1.0.0-py2.7.egg/mopidy_gmusic/actor.py", line 78, in _refresh_library
    self.playlists.refresh()
  File "/usr/local/lib/python2.7/dist-packages/Mopidy_GMusic-1.0.0-py2.7.egg/mopidy_gmusic/playlists.py", line 50, in refresh
    for track in self.backend.session.get_thumbs_up_songs():
  File "/usr/local/lib/python2.7/dist-packages/Mopidy_GMusic-1.0.0-py2.7.egg/mopidy_gmusic/session.py", line 71, in get_thumbs_up_songs
    return self.api.get_thumbs_up_songs()
AttributeError: 'Mobileclient' object has no attribute 'get_thumbs_up_songs'

**Comment by jonsxaero** [comment]
In commit https://github.com/simon-weber/Unofficial-Google-Music-API/commit/6313029904e39d8aa69b4d7bbe8c13c7d2c179cc , "get_thumbs_up_songs" method is changed by "get_promoted_songs" because according to weber "it actually gets some random 'promoted' tracks, which are likely to be up-rated". When I have some time, I will make a PR to try to resolve this issue.

**Comment by Cysioland** [comment]
@jonsxaero This affects me too. Can't wait for your fix.
